### PR TITLE
[tizen_app_manager] Add tizen_interop dependency

### PR DESCRIPTION
**Pull request by bbrto21**
Signed-off-by: Boram Bae <boram21.bae@samsung.com>

### CODE REVIEW

File: packages/tizen_app_manager/lib/src/app_context.dart [hunk 1]
```diff
@@ -41,204 +17,118 @@ class _AppManager {
...
-            'app_context_get_package_id')
-        .asFunction();
-    getProcessId = libAppMananger
-        .lookup<NativeFunction<_AppContextGetPid>>('app_context_get_pid')
-        .asFunction();
-    getAppState = libAppMananger
-        .lookup<NativeFunction<_AppContextGetAppState>>(
-            'app_context_get_app_state')
-        .asFunction();
+    // Since app_manager_terminate_app is non public API, load it using ffi directly.
```
Comment (bbrto21) [inline packages/tizen_app_manager/lib/src/app_context.dart:20]: @WonyoungChoi I can't get app_manager_terminate_app from tizen_interop, so I had to leave it as is.

File: packages/tizen_app_manager/lib/src/app_context.dart [hunk 2]
```diff
@@ -41,204 +17,118 @@ class _AppManager {
...
-
-    if (handleAddress == 0) {
-      final Pointer<Pointer<_ContextHandle>> pHandle = malloc();
-      final Pointer<Utf8> pAppId = appId.toNativeUtf8();
-      try {
-        final int ret = _appManager.getAppContext(pAppId, pHandle);
+    using((Arena arena) {
+      if (handleAddress == 0) {
+        final Pointer<Pointer<app_context_s>> handle = arena();
+        final Pointer<Char> pAppId = appId.toNativeChar(allocator: arena);
```
Comment (swift-kim) [inline packages/tizen_app_manager/lib/src/app_context.dart:37]: Please use a consistent naming style (either with or without leading `p`).

File: packages/tizen_app_manager/lib/src/app_context.dart [hunk 3]
```diff
@@ -41,204 +17,118 @@ class _AppManager {
...
-        .lookup<NativeFunction<_AppContextGetPid>>('app_context_get_pid')
-        .asFunction();
-    getAppState = libAppMananger
-        .lookup<NativeFunction<_AppContextGetAppState>>(
-            'app_context_get_app_state')
-        .asFunction();
+    // Since app_manager_terminate_app is non public API, load it using ffi directly.
     terminateApp = libAppMananger
-        .lookup<NativeFunction<_AppManagerTerminate>>(
+        .lookup<NativeFunction<Int32 Function(Pointer<app_context_s>)>>(
```
[the remark this answers is not in the document]
Comment (bbrto21) [reply packages/tizen_app_manager/lib/src/app_context.dart]: Oh thank you, I didn't know about them.

File: packages/tizen_app_manager/CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,8 @@
+## 0.2.0
+
+* Migrate FFI implementation to tizen_interop.
+* Remove `AppRunningContext.dispose`.
```
Comment (swift-kim) [inline packages/tizen_app_manager/CHANGELOG.md:4]: Please also add

```markdown
* Move `app_manager.dart` into the `src` directory.
```

**Review by swift-kim — APPROVED** [review]

File: packages/tizen_app_manager/CHANGELOG.md (hunk 4 above)
Comment (bbrto21) [reply packages/tizen_app_manager/CHANGELOG.md:4]: Done!

**Pull request merged**